modules/programs/waybar.nix

```diff
@@ -0,0 +1,371 @@
+
+      withSwayIntegration = mkOption {
+        type = bool;
+        default = false;
+        example = true;
+        description =
+          "Whether to bind the systemd service to the sway-session.target instead of graphical-session.target.";
+      };
+    };
+
+    style = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        CSS style of the bar.
+        See <link xlink:href="https://github.com/Alexays/Waybar/wiki/Configuration/>"/> for the documentation.
+      '';
+      example = ''
+        * {
+          border: none;
+          border-radius: 0;
+          font-family: Source Code Pro;
+        }
+        window#waybar {
+          background: #16191C;
+          color: #AAB2BF;
+        }
+        #workspaces button {
+          padding: 0 5px;
+        }
+      '';
+    };
+  };
+
+  config = let
+    # Inspired by https://github.com/NixOS/nixpkgs/pull/89781
+    writePrettyJSON = name: x:
+      pkgs.runCommandNoCCLocal name { } ''
+        echo '${builtins.toJSON x}' | ${pkgs.jq}/bin/jq . > $out
+      '';
+
+    configSource = let
+      # Removes nulls because Waybar ignores them for most values
+      removeNulls = filterAttrs (_: v: !isNull v);
+      # Makes the actual valid configuration Waybar accepts (strips our custom settings before converting to JSON)
+      makeConfiguration = configuration:
+        let
+          # The "modules" option is not valid in the JSON as its descendants have to live at the top-level
+          settingsWithoutModules =
+            filterAttrs (n: _: n != "modules") configuration;
+          settingsModules =
+            optionalAttrs (configuration.modules != { }) configuration.modules;
+        in removeNulls (settingsWithoutModules // settingsModules);
+      # The clean list of configurations
+      finalConfiguration = map makeConfiguration cfg.settings;
+    in writePrettyJSON "waybar-config.json" finalConfiguration;
+
+    warnings = let
+      mkPath = idx:
+        let i = toString idx;
+        in "${modulesPath}[definition ${i}-entry ${i}]";
+      mkUnreferencedModuleWarning = idx: name:
+        "The module '${name}' defined in '${mkPath idx}' is not referenced "
+        + "in either `modules-left`, `modules-center` or `modules-right` of Waybar's options";
+      mkUndefinedModuleWarning = idx: name:
+        "The module '${name}' defined in '${mkPath idx}' is neither "
+        + "a default module or a custom module declared in '${
+          mkPath idx
+        }.modules'";
+
+      # Find all modules in `modules-{left,center,right}` and `modules` not declared/referenced.
+      # cfg.settings is a list of Waybar configurations, we need to preserve the index for appropriate warnings
+      allFaultyModules = flip imap1 cfg.settings (idx: settings:
+        let
+          allModules = concatMap (x: attrByPath [ x ] [ ] settings) [
+            "left"
+            "center"
+            "right"
```

Comment:
Seems to be missing the `modules-` prefix for these attributes to be looked up.

Reply:
Thanks, fixed in the last commit